geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHincrByFloatIntegrationTest.java

```diff
@@ -0,0 +1,202 @@
+    assertThatThrownBy(() -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT))
+        .hasMessageContaining("ERR wrong number of arguments for 'hincrbyfloat' command");
+  }
+
+  @Test
+  public void givenHashNotProvided_returnsWrongNumberOfArgumentsError() {
+    assertThatThrownBy(() -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT, "key"))
+        .hasMessageContaining("ERR wrong number of arguments for 'hincrbyfloat' command");
+  }
+
+  @Test
+  public void givenIncrementNotProvided_returnsWrongNumberOfArgumentsError() {
+    assertThatThrownBy(() -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT, "key", "hash"))
+        .hasMessageContaining("ERR wrong number of arguments for 'hincrbyfloat' command");
+  }
+
+  @Test
+  public void givenMoreThanFourArgumentsProvided_returnsWrongNumberOfArgumentsError() {
+    assertThatThrownBy(
+        () -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT, "key", "hash", "5", "extraArg"))
+            .hasMessageContaining("ERR wrong number of arguments for 'hincrbyfloat' command");
+  }
+
+  @Test
+  public void testHincrByFloat_withInfinityAndVariants() {
+    jedis.hset("key", "number", "1.4");
+
+    assertThatThrownBy(
+        () -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT, "key", "number", "+inf"))
+            .hasMessage("ERR increment would produce NaN or Infinity");
+
+    assertThatThrownBy(
+        () -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT, "key", "number", "-inf"))
+            .hasMessage("ERR increment would produce NaN or Infinity");
+
+    assertThatThrownBy(
+        () -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT, "key", "number", "inf"))
+            .hasMessage("ERR increment would produce NaN or Infinity");
+
+    assertThatThrownBy(
+        () -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT, "key", "number", "+infinity"))
+            .hasMessage("ERR increment would produce NaN or Infinity");
+
+    assertThatThrownBy(
+        () -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT, "key", "number", "-infinity"))
+            .hasMessage("ERR increment would produce NaN or Infinity");
+
+    assertThatThrownBy(
+        () -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT, "key", "number", "infinity"))
+            .hasMessage("ERR increment would produce NaN or Infinity");
+
+    assertThatThrownBy(
+        () -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT, "key", "number", "nan"))
+            .hasMessage("ERR value is not a valid float");
+
+    assertThatThrownBy(
+        () -> jedis.sendCommand(Protocol.Command.HINCRBYFLOAT, "key", "number", "infant"))
+            .hasMessage("ERR value is not a valid float");
+  }
+
+  @Test
+  public void testHincrByFloat() {
+    double incr = 0.37373;
+    String key = "key";
+    String field = "field";
+    jedis.hset(key, field, "1.0");
+
+    double response = jedis.hincrByFloat(key, field, incr);
+    assertThat(response).isEqualTo(incr + 1, offset(.00001));
```

Comment:
I think I remember us talking about this at some point, but remind me again why we need the offset?

Reply:
Redis' floating point precision is not precise.